webserver.cpp

```diff
@@ -1037,18 +1054,28 @@ int http_parse_multipart_body(struct webserver_t *client, unsigned char *buf, ui
                 }
               }
               client->buffer[pos] = '=';
+              if(ending == 1) {
+                client->buffer[pos+1] = '\r';
+              }
               if(ending == 2) {
                 client->buffer[pos+1] = '\r';
                 client->buffer[pos+2] = '\n';
               }
+              if(ending == 3) {
+                client->buffer[pos+1] = '\r';
+                client->buffer[pos+2] = '\n';
+                client->buffer[pos+3] = '\r';
+                if (ending3dash) client->buffer[pos+3] = '-';
+              }
               client->readlen += ((client->ptr-(pos+1))-ending);
               client->ptr = pos+1+ending;
+	      loop = 0;
```

Comment:
This extra loop=0 saves an extra while loop run. The client buffer is just emptied into the callback so new data should be received (if available). Without this loop=0 the while loop will run again to step 7 and take the loop=0 on line 1078